ui/dashboards/src/components/Variables/variable-model.ts

```diff
@@ -101,3 +101,25 @@ export const VARIABLE_TYPES = [
   { label: 'List', kind: 'ListVariable' },
   { label: 'Text', kind: 'TextVariable' },
 ] as const;
+
+/*
+ * Check whether saved variable definitions are out of date with current default list values in Zustand store
+ */
+export function checkSavedVariablesStatus(definitions: VariableDefinition[], varState: VariableStateMap) {
+  let isSavedVariablesOutdated = false;
+  definitions.forEach((saveVariable) => {
+    if (saveVariable.kind === 'ListVariable') {
+      const currentVariable = varState[saveVariable.spec.name];
+      if (saveVariable.spec.default_value !== currentVariable?.value) {
+        isSavedVariablesOutdated = true;
```

Comment:
i think we can just return true here once we know it's outdated

Reply:
Yeah lol whoops, I’ll update this, before I clarified with Mohit, I was keeping track of two booleans in this function but he said it was fine to show the dialog after a new variable is created